### PR TITLE
Weaviate config lam

### PR DESCRIPTION
**Pull request by Lamp-Nguyen**

### CODE REVIEW

File: backend/index.ts [hunk 1]
```diff
@@ -1 +1,24 @@
+import * as dotenv from 'dotenv';
+import { weaviateRoute } from './weaviateRoute'; 
+
+dotenv.config;
+
+const weaviate_key = process.env.WEAVIATE_API_KEY ?? '';
+const cohere_key = process.env.COHERE_API_KEY ?? '';
+
+if (!weaviate_key) {
+    console.log("API KEY WAS NOT PROVIDED");
```
Comment (duong-vo) [inline backend/index.ts:10]: throw error instead of console log

File: backend/index.ts [hunk 2]
```diff
@@ -1 +1,24 @@
...
+const testStr2 = "Often an urgent rather than an Emergency health problem exists. An urgent health problem is an unexpected illness or injury that calls for care that cannot wait until a regularly scheduled office visit. Urgent health problems are not life threatening and do not call for the use of an Emergency room. Urgent health problems include earache, sore throat and fever (not above one-hundred and four (104) degrees). Benefits for Urgent Care may include: · X-ray services · Care for broken bones · Tests such as flu, urinalysis, pregnancy test, rapid strep"
+//await weaviateClient.initSchema();
+//await weaviateClient.deleteSchema('EOC');
+//await weaviateClient.addClassObj(testStr1);
+//await weaviateClient.addClassObj(testStr2);
+//console.log(JSON.stringify(await weaviateClient.getSchema(), null, 2));
+const res = await weaviateClient.generativeQuery('Urgent care');
+//console.log(res);
+
 console.log("Hello via Bun!");
```
Comment (duong-vo) [inline backend/index.ts:24]: remove this

File: backend/weaviateRoute.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            scheme: 'https',
+            host: 'eoc-cluster-nl8w5r28.weaviate.network',
+            apiKey: new ApiKey(apiKey),
+            headers: { 'X-Cohere-Api-Key': cohereKey },
+        })
+    }
+
+    // Return the current class schema as a WeaviateClass
+    async getSchema() : Promise<WeaviateClass> {
+        var res = await this.client.schema.getter().do() as WeaviateClass;
```
Comment (duong-vo) [inline backend/weaviateRoute.ts:21]: use const

File: backend/weaviateRoute.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        return res;
+    }
+
+    // Initialize a class schema using the provided classDefinition.ts
+    // if the schema doesn't already exists.
+    async initSchema() {
+        var curSchema = await this.client.schema.getter().do();
```
Comment (duong-vo) [inline backend/weaviateRoute.ts:28]: use const

File: backend/weaviateRoute.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    }
+
+    // Delete the class schema with the provided className
+    async deleteSchema(className: string) {
+        try {
+            await this.client.schema.classDeleter().withClassName(className).do();
+            console.log('Schema deleted successfully');
+        } catch (error) {
+            console.log(error);
```
Comment (duong-vo) [inline backend/weaviateRoute.ts:43]: throw error

File: backend/weaviateRoute.ts (hunk 4 above)
Comment (Lamp-Nguyen) [reply backend/weaviateRoute.ts:28]: you a bitch

File: backend/weaviateRoute.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        }
+    }
+
+    // Add a class object to the cluster
+    async addClassObj(info: string) {
+        let batcher: ObjectsBatcher = this.client.batch.objectsBatcher();
```
Comment (duong-vo) [inline backend/weaviateRoute.ts:49]: use const

**Review by dominhnhut01 — APPROVED** [review]
lgtm

**Pull request merged**